product_pricelist_direct_print/wizards/product_pricelist_print.py

```diff
@@ -0,0 +1,217 @@
+            if len(items.mapped("pricelist_id")) == 1:
+                res["pricelist_id"] = items[0].pricelist_id.id
+            product_items = items.filtered(
+                lambda x: x.applied_on == "0_product_variant"
+            )
+            template_items = items.filtered(lambda x: x.applied_on == "1_product")
+            category_items = items.filtered(
+                lambda x: x.applied_on == "2_product_category"
+            )
+            # Convert al pricelist items to their affected variants
+            if product_items:
+                res["show_variants"] = True
+                product_ids = product_items.mapped("product_id")
+                product_ids |= template_items.mapped(
+                    "product_tmpl_id.product_variant_ids"
+                )
+                product_ids |= product_ids.search(
+                    [
+                        ("sale_ok", "=", True),
+                        ("categ_id", "in", category_items.mapped("categ_id").ids),
+                    ]
+                )
+                res["product_ids"] = [(6, 0, product_ids.ids)]
+            # Convert al pricelist items to their affected templates
+            if template_items and not product_items:
+                product_tmpl_ids = template_items.mapped("product_tmpl_id")
+                product_tmpl_ids |= product_tmpl_ids.search(
+                    [
+                        ("sale_ok", "=", True),
+                        ("categ_id", "in", category_items.mapped("categ_id").ids),
+                    ]
+                )
+                res["product_tmpl_ids"] = [(6, 0, product_tmpl_ids.ids)]
+            # Only category items, we just set the categories
+            if category_items and not product_items and not template_items:
+                res["categ_ids"] = [(6, 0, category_items.mapped("categ_id").ids)]
+        return res
+
+    def print_report(self):
+        if not (
+            self.pricelist_id
+            or self.partner_count
+            or self.show_standard_price
+            or self.show_sale_price
+        ):
+            raise ValidationError(
+                _(
+                    "You must set price list or any customer "
+                    "or any show price option."
+                )
+            )
+        return self.env.ref(
+            "product_pricelist_direct_print." "action_report_product_pricelist"
+        ).report_action(self)
+
+    def action_pricelist_send(self):
+        self.ensure_one()
+        if self.partner_count > 1:
+            self.send_batch()
+        else:
+            if self.partner_count == 1:
+                partner = self.partner_ids[0]
+                self.write(
+                    {
+                        "partner_id": partner.id,
+                        "pricelist_id": partner.property_product_pricelist.id,
+                    }
+                )
+            return self.message_composer_action()
```

Comment:
Can reduce complexity...
```suggestion
            self.send_batch()
            return
        if self.partner_count == 1:
            partner = self.partner_ids[0]
            self.write(
                {
                    "partner_id": partner.id,
                    "pricelist_id": partner.property_product_pricelist.id,
                }
            )
        return self.message_composer_action()
```